src/fallback/dom_parser_implementation.cpp

```diff
@@ -128,16 +145,26 @@ really_inline error_code scan() {
         break;
     }
   }
-  if (unlikely(next_structural_index == parser.structural_indexes.get())) {
-    return EMPTY;
-  }
   *next_structural_index = len;
-  next_structural_index++;
   // We pad beyond.
   // https://github.com/simdjson/simdjson/issues/906
-  next_structural_index[0] = len;
-  next_structural_index[1] = 0;
+  *(next_structural_index+1) = len;
```

Comment:
Here you can leave things are you wrote them but I feel compelled to say that  I personally find that the syntax...
```C++
next_structural_index[0] = len;
next_structural_index[1] = 0;
next_structural_index[2] = 0;
```
easier on the eyes than 
```
*next_structural_index = len;
*(next_structural_index+1) = 0;
*(next_structural_index+2) = 0;
```

( I don't really care though.)

Reply:
Sure, I can do that. The main reason I did this kind of thing is I didn't want people thinking next_structural_index was actually an array starting at next_structural_index; but I think that's pretty obvious from the name.

I might keep *next_structural_index = len; just to make it look like all the other uses, but the array syntax looks way nicer for the other two.